app/controllers/steps/supporting_evidence_controller.rb

```diff
@@ -3,43 +3,41 @@
 module Steps
   class SupportingEvidenceController < Steps::BaseStepController
     skip_before_action :verify_authenticity_token
-    before_action :supporting_evidence
+    before_action :supporting_evidence, :file_uploader
+
+    SUPPORTED_FILE_TYPES = %w[application/msword application/vnd.openxmlformats-officedocument.wordprocessingml.document
+                              application/rtf image/jpeg image/bmp image/png image/tiff application/pdf].freeze
+
     def edit
       @form_object = SupportingEvidenceForm.build(
         current_application
       )
     end
 
     def create
-      evidence = SupportingEvidence.create(
-        file_name: params[:documents].original_filename,
-        file_type: params[:documents].content_type,
-        file_size: params[:documents].tempfile.size,
-        claim: current_application,
-        file: params[:documents]
-      )
+      unless SUPPORTED_FILE_TYPES.include? params[:documents].content_type
+        return return_error(nil, { message: 'Incorrect file type provided' })
+      end
 
-      render json: {
-        success: {
-          evidence_id: evidence.id
-        }
-      }, status: :ok
+      file_path = @file_uploader.upload(params[:documents])
+      evidence = save_evidence_data(params[:documents], file_path)
+      return_success({ evidence_id: evidence.id, file_name: params[:documents].original_filename })
+    rescue StandardError => e
+      return_error(e, { message: 'Unable to upload file at this time' })
     end
 
     def update
       update_and_advance(SupportingEvidenceForm, as: :supporting_evidence)
     end
 
     def destroy
-      SupportingEvidence.destroy(params[:evidence_id])
-      render json: {
-        success: true
-      }
+      evidence = SupportingEvidence.find_by(id: params[:evidence_id])
```

Comment:
prefer to use `claim.supporting_evidences.find_by` as this ensures scope (i.e. can't delete someone else evidence)

Reply:
👍